invoke/context.py

```diff
@@ -192,6 +202,134 @@ def sudo(self, command, **kwargs):
             else:
                 raise
 
+    def _prefix_commands(self, command):
+        """
+        Prefixes ``command`` with all prefixes found in ``command_prefixes``.
+
+        ``command_prefixes`` is a list of strings which is modified by the
+        `prefix` context manager.
+        """
+        prefixes = list(self.command_prefixes)
+        current_directory = self.cwd
+        if current_directory:
+            prefixes.insert(0, 'cd %s' % current_directory)
+
+        return ' && '.join(prefixes + [command])
+
+    @contextmanager
+    def prefix(self, command):
+        """
+        Prefix all nested `run`/`sudo` commands with given command plus ``&&``.
+
+        Most of the time, you'll want to be using this alongside a shell script
+        which alters shell state, such as ones which export or alter shell
+        environment variables.
+
+        For example, one of the most common uses of this tool is with the
+        ``workon`` command from `virtualenvwrapper
+        <http://www.doughellmann.com/projects/virtualenvwrapper/>`_::
```

Comment:
This is now https://virtualenvwrapper.readthedocs.io so we should probs update it!